compute_endpoint/globus_compute_endpoint/endpoint/endpoint_manager.py

```diff
@@ -212,6 +219,38 @@ def wait_for_children(self):
                     log.warning(
                         f"Command terminated by signal: {-rc} ({pid}){proc_args}"
                     )
+
+                ep_name = "<unknown>"
+                for p, n in list(self._running_eps):
+                    if p == pid:
+                        ep_name = n
+                        self._running_eps.discard((p, n))
```

Comment:
Is it possible for a running process to have two endpoint names associated with it?

If not (and the process ID is guaranteed to be unique) it may be useful to use a dict instead of a set as the `._running_eps` data type. That would eliminate this loop:

```suggestion
                ep_name = self._running_eps.pop(pid, "<unknown>")
```

There's a loop below that searches by endpoint name; that would need to loop over `._running_eps.items()`. :+1:

Reply:
Following an out-of-band conversation with Kevin, I just pushed an update that unifies the various ad-hoc data structures in play here into a `_children` dict, see here: https://github.com/funcx-faas/funcX/compare/04cb568476fc1a328977a1a5fe60aa0deb33ca65..d19796ae1f6f1a531b4ecc167c418752f037ef59?diff=split